src/Instrument/FileSystem/Enumerator.php

```diff
@@ -78,24 +79,29 @@ public function enumerate()
     public function getFilter()
     {
         $rootDirectory = $this->rootDirectory;
-        $includePaths  = $this->includePaths;
-        $excludePaths  = $this->excludePaths;
+        $includePaths = $this->includePaths;
+        $excludePaths = $this->excludePaths;
+
+        // Parse wildcard paths
+        $excludePaths = array_map([$this, 'parseWildcardPath'], $excludePaths);
+        $includePaths = array_map([$this, 'parseWildcardPath'], $includePaths);
 
-        return function(\SplFileInfo $file) use ($rootDirectory, $includePaths, $excludePaths) {
+        return function (\SplFileInfo $file) use ($rootDirectory, $includePaths, $excludePaths) {
             if ($file->getExtension() !== 'php') {
                 return false;
             };
 
-            $realPath = $file->getRealPath();
+            $fullPath = $file->getRealPath();
+            $filePath = $file->getPath();
             // Do not touch files that not under rootDirectory
-            if (strpos($realPath, $rootDirectory) !== 0) {
+            if (strpos($fullPath, $rootDirectory) !== 0) {
                 return false;
             }
 
             if (!empty($includePaths)) {
                 $found = false;
-                foreach ($includePaths as $includePath) {
-                    if (strpos($realPath, $includePath) === 0) {
+                foreach ($includePaths as $includePattern) {
+                    if (preg_match('%' . $includePattern . '%', $filePath)) {
```

Comment:
Problem is that fnmatch does not work with multiple subfolders. so `/base/*/test` only includes `/base/sub/test` but not `/base/sub/sub/test`